gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonClientCallImpl.java

```diff
@@ -161,6 +168,36 @@ public void start(Listener<ResponseT> responseListener, HttpJsonMetadata request
       this.listener = responseListener;
       this.requestHeaders = requestHeaders;
     }
+
+    // Use the timeout duration value instead of calculating the future Instant
+    Duration timeout = callOptions.getTimeout();
+    if (timeout != null) {
+      // If the future timeout amount has been calculated as a negative value,
+      // we cancel the call immediately as the deadline has already been exceeded.
+      // The RetryAlgorithm should for this value and not run schedule a run
+      // if this is negative.
+      long timeoutNanos = 0;
+      if (!timeout.isNegative()) {
+        timeoutNanos = timeout.toNanos();
+      }
+      this.deadlineCancellationExecutor.schedule(
+          this::closeAndNotifyListeners, timeoutNanos, TimeUnit.NANOSECONDS);
+    }
+  }
+
+  // No need to trigger the deliver() loop again as we have already closed the runnable
+  // task and added the OnCloseNotificationTask. We notify the FutureListener that the
+  // there is a timeout exception from this RPC call (DEADLINE_EXCEEDED)
+  private void closeAndNotifyListeners() {
+    synchronized (lock) {
+      close(
```

Comment:
This [comment](https://github.com/googleapis/gapic-generator-java/blob/4bb0a5e3ce78878841e87a47cd457995ea9ed87e/gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonClientCallImpl.java#L393-L396) in `close` method makes me worried that we may still have to wait for the server to fully send the response.

Reply:
Yeah the cancellation in the runnable doesn't seem like it's actually cancel's the runnable. I'll look into the possibility of being able to actually disconnect the connection instead of waiting for the socket timeout.